onnx_coreml/_operators_nd.py

```diff
@@ -16,6 +16,107 @@
 
 INT_MAX = 2**30
 
+def _convert_concat(builder, node, graph, err):
+    '''
+    convert to CoreML ConcatND Layer:
+    https://github.com/apple/coremltools/blob/655b3be5cc0d42c3c4fa49f0f0e4a93a26b3e492/mlmodel/format/NeuralNetwork.proto#L3521
+    '''
+
+    axis = node.attrs.get('axis')
+
+    for i in range(len(node.inputs)):
+        if node.inputs[i] in node.input_tensors:
+            builder.add_load_constant_nd(
```

Comment:
What if these tensors have already been added by some other layer as "load constants" ? Think of a network like:

input: x
a =load_constant
b = x + a
c = concat(b,a)

I wonder if we could maybe maintain a dict somewhere of all the load constant nodes added to the CoreML spec and only add it again if it has not been added yet.

Reply:
sounds good. we can maintain a set of nodes added and if node is present then we re-use it.